internal/pipeline/executor.go

```diff
@@ -190,8 +190,8 @@ func (p *pkg) localResources(includeMetadata bool) (resources []*yaml.RNode, err
 		err = fmt.Errorf("failed to read resources for pkg %s %w", p.Path(), err)
 		return resources, err
 	}
+	resources = filterFnConfig(resources)
```

Comment:
I am still thinking the correct place to do it, but the intent is if there are resources that are provided as input to a function in the pipeline, we want to exclude such resources. So this will require reading the function config file and fetch the GVKNN (could be multiple) and filter those out. Basically, treat those function config as input to the pipeline, so pipeline shouldn't be applied to them.

Filtering Configmap on the basis of annotation `config.kubernetes.io/function` will filter our old style function definitions. I missed that, we need that filtering as well.

Reply:
resolving this conversation for now because such filtering will be included on a different PR if we decide to do it